forc-plugins/forc-node/src/util.rs

```diff
@@ -0,0 +1,166 @@
+use crate::consts::DB_FOLDER;
+use anyhow::{anyhow, bail, Result};
+use dialoguer::{theme::ColorfulTheme, Confirm, Input, Password};
+use forc_util::user_forc_directory;
+use semver::Version;
+use serde::{Deserialize, Serialize};
+use std::{
+    fmt::Display,
+    path::PathBuf,
+    process::{Command, Stdio},
+};
+
+pub enum DbConfig {
+    Local,
+    Testnet,
+    Ignition,
+}
+
+impl From<DbConfig> for PathBuf {
+    fn from(value: DbConfig) -> Self {
+        let user_db_dir = user_forc_directory().join(DB_FOLDER);
+        match value {
+            DbConfig::Local => user_db_dir.join("local"),
+            DbConfig::Testnet => user_db_dir.join("ignition-testnet"),
+            DbConfig::Ignition => user_db_dir.join("ignition"),
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct KeyPair {
+    pub peer_id: String,
+    pub secret: String,
```

Comment:
nit: may not want to expose publicly (taking inspiration from `WalletUnlocked`)
```suggestion
    secret: String,
```

Reply:
This is contained in forc-node, while WalletUnlocked is exposed to other crates and thus encapsulation is needed. Here I am saving some extra getter code as this is simply a short lived, pass through type.